src/main/kotlin/me/serce/solidity/ide/navigation/SolContractImplementationSearcher.kt

```diff
@@ -68,7 +69,14 @@ private fun findAllImplementationsInAction(
 }
 
 fun SolContractDefinition.findImplementations(): Query<SolContractDefinition> {
-  val solOnlyScope = this.useScope.intersectWith(FilesScope.getScopeRestrictedByFileTypes(this.resolveScope, SolidityFileType))
+  val virtualFile = containingFile.virtualFile
+  val resolveScope = when {
+    // When the file for which we're performing search for usages is a part of the sources/libs in the current project
+    virtualFile != null && resolveScope.contains(virtualFile) -> resolveScope
+    // However, if the project doesn't store solidity files in its source folder, we perform a global search.
+    else -> allScope(project)
```

Comment:
cc @mbolotov, it seems that the change in https://github.com/intellij-solidity/intellij-solidity/pull/392 affected go-to-impl in the cases where the file isn't a part of sources which can be the case in many cases where solidity files are stored outside of folders that are marked as "sources".